qiskit/circuit/quantumcircuit.py

```diff
@@ -2205,32 +2209,33 @@ def cu(self, theta, phi, lam, gamma,   # pylint: disable=invalid-name
         return self.append(CUGate(theta, phi, lam, gamma, label=label, ctrl_state=ctrl_state),
                            [control_qubit, target_qubit], [])
 
-    def u1(self, theta, qubit):  # pylint: disable=invalid-name
+    def u1(self, theta, qubit, label=None):  # pylint: disable=invalid-name
         """Apply :class:`~qiskit.circuit.library.U1Gate`."""
         from .library.standard_gates.u1 import U1Gate
-        return self.append(U1Gate(theta), [qubit], [])
+        return self.append(U1Gate(theta, label=label), [qubit], [])
 
     def cu1(self, theta, control_qubit, target_qubit, label=None, ctrl_state=None):
         """Apply :class:`~qiskit.circuit.library.CU1Gate`."""
         from .library.standard_gates.u1 import CU1Gate
         return self.append(CU1Gate(theta, label=label, ctrl_state=ctrl_state),
                            [control_qubit, target_qubit], [])
 
-    def mcu1(self, lam, control_qubits, target_qubit):
+    def mcu1(self, lam, control_qubits, target_qubit, label=None, ctrl_state=None):
         """Apply :class:`~qiskit.circuit.library.MCU1Gate`."""
         from .library.standard_gates.u1 import MCU1Gate
         num_ctrl_qubits = len(control_qubits)
-        return self.append(MCU1Gate(lam, num_ctrl_qubits), control_qubits[:] + [target_qubit], [])
+        return self.append(MCU1Gate(lam, num_ctrl_qubits, label=label, ctrl_state=ctrl_state),
+                           control_qubits[:] + [target_qubit], [])
 
-    def u2(self, phi, lam, qubit):  # pylint: disable=invalid-name
+    def u2(self, phi, lam, qubit, label=None):  # pylint: disable=invalid-name
```

Comment:
isn't None the default?

Reply:
It is for the underlying U2Gate class, but the issue this PR is resolving is that there was no interface to explicitly pass a label or ctrl_state parameter when using the quantumcircuit methods. I added default parameters for each of the gates so that you can create a label for the added gate if desired.

Reply:
These tests are to make sure the current behavior holds (aka, the default). If you want to test the new param, create tests with `u2(self, phi, lam, qubit, label='pink fluffy unicorns')`. But do not modify these one :)

Reply:
oh! I thought you were changing the tests... sorry!